aws/src/main/python/instanceNotifier/instance_notifier.py

```diff
@@ -0,0 +1,226 @@
+    conn = http.client.HTTPSConnection('hooks.slack.com')
+    conn.request('POST', slack_hook, json.dumps(payload), headers)
+    response = conn.getresponse()
+    logger.info('Received response from slack notification for attempted instance use: ' + response.read().decode())
+
+def stop_instance(instance_id):
+    logger.info(f"Stopping {instance_id}")
+    ec2_client.stop_instances(InstanceIds=[instance_id])
+
+def set_override_tag_to_empty_for_(instance_id):
+    ec2_client.create_tags(
+    Resources=[instance_id],
+    Tags=[
+        {
+            'Key': 'BudgetOverride',
+            'Value': '',
+        },
+    ],
+)
+
+# Expected format: {"instance-id": <instance-id>, "state": <state>, "region": <region>, "account-id": <account-id>}
+def lambda_handler(event, context):
+    state = event.get('state')
+    if state != "running":
+        return "Instance state change of {state} is not supported for this lambda - only 'running' is considered".format(state=state)
+    region = event.get('region')
+    instance_id = event.get('instance-id')
+    account_id = event.get('account-id')
+
+    allowed_instances = os.environ['ALLOWED_INSTANCES']
+    if instance_id in allowed_instances:
+        logger.info(f"Allowing {instance_id} to run")
+        return "Done"
+
+    init_budget_client()
+    budget_response = budget_client.describe_budget(
+        AccountId=account_id,
+        BudgetName=os.environ["BUDGET_NAME"]
+    )
+
+    budget_details = budget_response.get('Budget')
+    budget_limit_amount = Decimal(budget_details.get('BudgetLimit').get('Amount'))
+    current_spent_amount = Decimal(budget_details.get('CalculatedSpend').get('ActualSpend').get('Amount'))
+    percent_of_budget_used = (current_spent_amount / budget_limit_amount) * 100
+
+    # 150%+ require tag - notify slack if blocked
+    if percent_of_budget_used <= 150:
+        return "Budget is not over 150%. Stopping check."
+
+    today = date.today()
+    current_day = Decimal(today.strftime("%d"))
+    current_month = today.strftime("%m")
+    current_year = today.strftime("%Y")
+    days_in_current_month = monthrange(int(current_year), int(current_month))[1]
+    percent_of_month = (current_day / Decimal(days_in_current_month)) * 100
+
+    # LATER TODO: Add link to budget?
+    # LATER TODO: Add an override option that is not emptied (true_always?) or for # of starts (true_3? - decrement and update tag)
+    # LATER TODO: Add a link to trigger an automatic approval. Maybe link to instance is enough, since can tag there?
+    # LATER TODO: Create a report of all usages and cost and send as a daily/weekly? email
+    init_ec2_client(region)
+    instance_details = get_instance_details_for_(instance_id)
+
+    instance_stopped = False
+    if percent_of_budget_used >= 300:
+        stop_instance(instance_id)
+        instance_stopped = True
+    elif percent_of_budget_used >= 150 and instance_details.budget_override != "true":
```

Comment:
In the `start_instance` method of the lambda we set the tag value 'True"
```
           Tags=[
                {
                    'Key': 'BudgetOverride',
                    'Value': 'True'
                }
            ]
```
But it is compared to 'true' here. Is this OK?

Reply:
Changing it to a lowercase so it handles all flavors of casing